commit f1d40433352e5d4babd59c0dd50b5f9414073ddb
Author: Takashi Iwai <tiwai@suse.de>
Date:   Fri Jun 17 16:40:48 2022 +0200

    ALSA: rawmidi: Move lock to snd_rawmidi_substream
    
    Having a lock in snd_rawmidi_runtime can be a problem especially when
    a substream is accessed from the outside, as the runtime creation
    might be racy with the external calls.  As a first step for hardening,
    move the spinlock from snd_rawmidi_runtime to snd_rawmidi_substream.
    
    This patch just replaces the lock calls, no real functional change is
    put yet.
    
    Link: https://lore.kernel.org/r/20220617144051.18985-3-tiwai@suse.de
    Signed-off-by: Takashi Iwai <tiwai@suse.de>

diff --git a/include/sound/rawmidi.h b/include/sound/rawmidi.h
index 9402c25ae9ba..e1f59b2940af 100644
--- a/include/sound/rawmidi.h
+++ b/include/sound/rawmidi.h
@@ -63,7 +63,6 @@ struct snd_rawmidi_runtime {
 	size_t xruns;		/* over/underruns counter */
 	int buffer_ref;		/* buffer reference count */
 	/* misc */
-	spinlock_t lock;
 	wait_queue_head_t sleep;
 	/* event handler (new bytes, input only) */
 	void (*event)(struct snd_rawmidi_substream *substream);
@@ -85,6 +84,7 @@ struct snd_rawmidi_substream {
 	unsigned int clock_type;	/* clock source to use for input framing */
 	int use_count;			/* use counter (for output) */
 	size_t bytes;
+	spinlock_t lock;
 	struct snd_rawmidi *rmidi;
 	struct snd_rawmidi_str *pstr;
 	char name[32];